internal/testhelpers/session.go

```diff
@@ -136,92 +178,261 @@ func NewHTTPClientWithSessionToken(t *testing.T, reg *driver.RegistryDefault, se
 	}
 }
 
-func NewHTTPClientWithArbitrarySessionToken(t *testing.T, reg *driver.RegistryDefault) *http.Client {
-	req := x.NewTestHTTPRequest(t, "GET", "/sessions/whoami", nil)
-	s, err := session.NewActiveSession(req,
-		&identity.Identity{ID: x.NewUUID(), State: identity.StateActive},
-		NewSessionLifespanProvider(time.Hour),
-		time.Now(),
-		identity.CredentialsTypePassword,
-		identity.AuthenticatorAssuranceLevel1,
-	)
-	require.NoError(t, err, "Could not initialize session from identity.")
+func EnsureAAL(t *testing.T, c *http.Client, ts *httptest.Server, aal string, methods ...string) {
+	res, err := c.Get(ts.URL + session.RouteWhoami)
+	require.NoError(t, err)
+	sess := x.MustReadAll(res.Body)
+	require.NoError(t, res.Body.Close())
+	assert.EqualValues(t, aal, gjson.GetBytes(sess, "authenticator_assurance_level").String())
+	for _, method := range methods {
+		assert.EqualValues(t, method, gjson.GetBytes(sess, "authentication_methods.#(method=="+method+").method").String())
+	}
+	assert.Len(t, gjson.GetBytes(sess, "authentication_methods").Array(), 1+len(methods))
+}
 
-	return NewHTTPClientWithSessionToken(t, reg, s)
+func NewAuthorizedTransport(t *testing.T, reg *driver.RegistryDefault, sess *session.Session) *x.TransportWithHeader {
+	maybePersistSession(t, reg, sess)
+
+	return x.NewTransportWithHeader(http.Header{
+		"Authorization": {"Bearer " + sess.Token},
+	})
 }
 
-func NewHTTPClientWithArbitrarySessionCookie(t *testing.T, reg *driver.RegistryDefault) *http.Client {
-	req := x.NewTestHTTPRequest(t, "GET", "/sessions/whoami", nil)
-	s, err := session.NewActiveSession(req,
-		&identity.Identity{ID: x.NewUUID(), State: identity.StateActive},
-		NewSessionLifespanProvider(time.Hour),
-		time.Now(),
-		identity.CredentialsTypePassword,
-		identity.AuthenticatorAssuranceLevel1,
+// A builder struct to help customize an authenticated http client
+type HTTPClientBuilder struct {
+	t        *testing.T
+	request  *http.Request
+	identity *identity.Identity
+	session  *session.Session
+}
```

Comment:
I opted to use the builder pattern for these clients because it gives a more flexibility interface while building the clients. With it, we can present the same interface that most consumers of these helpers were already using (using the wrappers at the bottom of this file), but in the rare cases where more customization is needed we can use the builder methods directly however we need.